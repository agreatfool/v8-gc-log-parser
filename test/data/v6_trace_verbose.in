[15368:0x102004600]     6314 ms: Scavenge 181.6 (213.4) -> 180.5 (224.4) MB, 7.3 / 0.0 ms [allocation failure].
[15368:0x102004600] Memory allocator,   used: 229828 KB, available: 1236540 KB
[15368:0x102004600] New space,          used:   3903 KB, available:  12216 KB, committed:  32768 KB
[15368:0x102004600] Old space,          used:  62965 KB, available:   6373 KB, committed:  70612 KB
[15368:0x102004600] Code space,         used:   1270 KB, available:     43 KB, committed:   3072 KB
[15368:0x102004600] Map space,          used:    267 KB, available:    802 KB, committed:   1104 KB
[15368:0x102004600] Large object space, used: 116408 KB, available: 1235499 KB, committed: 119200 KB
[15368:0x102004600] All spaces,         used: 184814 KB, available: 1254935 KB, committed: 226756 KB
[15368:0x102004600] External memory reported:     16 KB
[15368:0x102004600] Total time spent in GC  : 63.6 ms
[15368:0x102004600] Increasing marking speed to 3 due to high promotion rate
[15368:0x102004600] Heap growing factor 4.0 based on mu=0.970, speed_ratio=8 (gc=3103504, mutator=368116)
[15368:0x102004600] Grow: old size: 68596 KB, new limit: 290507 KB (4.0)
[15368:0x102004600]     8049 ms: Scavenge 406.0 (440.6) -> 404.3 (452.6) MB, 5.9 / 0.0 ms (+ 6.4 ms in 42 steps since last GC) [allocation failure].
[15368:0x102004600] Memory allocator,   used: 463460 KB, available: 1002908 KB
[15368:0x102004600] New space,          used:   7345 KB, available:   8774 KB, committed:  32768 KB
[15368:0x102004600] Old space,          used: 128565 KB, available:   9346 KB, committed: 140244 KB
[15368:0x102004600] Code space,         used:   1264 KB, available:     59 KB, committed:   3072 KB
[15368:0x102004600] Map space,          used:    267 KB, available:    802 KB, committed:   1104 KB
[15368:0x102004600] Large object space, used: 276568 KB, available: 1001867 KB, committed: 283200 KB
[15368:0x102004600] All spaces,         used: 414010 KB, available: 1020849 KB, committed: 460388 KB
[15368:0x102004600] External memory reported:     24 KB
[15368:0x102004600] Total time spent in GC  : 128.8 ms
[15368:0x102004600] Heap growing factor 4.0 based on mu=0.970, speed_ratio=11 (gc=4136271, mutator=377178)
[15368:0x102004600] Grow: old size: 136882 KB, new limit: 563650 KB (4.0)
[15368:0x102004600]     8149 ms: Mark-sweep 435.2 (480.9) -> 137.5 (184.8) MB, 6.4 / 0.0 ms (+ 7.6 ms in 112 steps since start of marking, biggest step 2.0 ms) [GC interrupt] [GC in old space requested].
[15368:0x102004600] Memory allocator,   used: 189252 KB, available: 1277116 KB
[15368:0x102004600] New space,          used:   3879 KB, available:  12240 KB, committed:  32768 KB
[15368:0x102004600] Old space,          used: 134586 KB, available:   9839 KB, committed: 148436 KB
[15368:0x102004600] Code space,         used:   1249 KB, available:      0 KB, committed:   3072 KB
[15368:0x102004600] Map space,          used:    265 KB, available:      0 KB, committed:   1104 KB
[15368:0x102004600] Large object space, used:    781 KB, available: 1276075 KB, committed:    800 KB
[15368:0x102004600] All spaces,         used: 140762 KB, available: 1298154 KB, committed: 186180 KB
[15368:0x102004600] External memory reported:     24 KB
[15368:0x102004600] Total time spent in GC  : 135.2 ms
[15368:0x102004600]     9874 ms: Scavenge 315.6 (357.3) -> 314.2 (366.3) MB, 6.7 / 0.0 ms [allocation failure].
[15368:0x102004600] Memory allocator,   used: 375108 KB, available: 1091260 KB
[15368:0x102004600] New space,          used:   5428 KB, available:  10691 KB, committed:  32768 KB
[15368:0x102004600] Old space,          used: 201516 KB, available:  13911 KB, committed: 219092 KB
[15368:0x102004600] Code space,         used:   1287 KB, available:    974 KB, committed:   3072 KB
[15368:0x102004600] Map space,          used:    267 KB, available:    802 KB, committed:   1104 KB
[15368:0x102004600] Large object space, used: 113283 KB, available: 1090219 KB, committed: 116000 KB
[15368:0x102004600] All spaces,         used: 321784 KB, available: 1116598 KB, committed: 372036 KB
[15368:0x102004600] External memory reported:     16 KB
[15368:0x102004600] Total time spent in GC  : 198.4 ms
[15368:0x102004600] Memory reducer: call rate 0.556, low alloc, foreground
[15368:0x102004600] Memory reducer: started GC #1
