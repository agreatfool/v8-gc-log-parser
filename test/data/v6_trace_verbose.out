[
  {
    "pid": 15368,
    "isolate": "0x102004600",
    "time_since_init": 6314,
    "data": {
      "collector": "scavenge",
      "start_object_size": 181.6,
      "end_object_size": 180.5,
      "start_memory_size": 213.4,
      "end_memory_size": 224.4,
      "pause": 7.3,
      "external_time": 0,
      "gc_reason": "allocation failure"
    },
    "type": "trace"
  },
  {
    "pid": 15368,
    "isolate": "0x102004600",
    "type": "verbose",
    "data": {
      "memory_allocator": {
        "used": 229828,
        "available": 1236540
      },
      "new_space": {
        "used": 3903,
        "available": 12216,
        "committed": 32768
      },
      "old_space": {
        "used": 62965,
        "available": 6373,
        "committed": 70612
      },
      "code_space": {
        "used": 1270,
        "available": 43,
        "committed": 3072
      },
      "map_space": {
        "used": 267,
        "available": 802,
        "committed": 1104
      },
      "large_object_space": {
        "used": 116408,
        "available": 1235499,
        "committed": 119200
      },
      "all_spaces": {
        "used": 184814,
        "available": 1254935,
        "committed": 226756
      },
      "external_memory_reported": 16,
      "total_time_spent_in_gc": 63.6
    }
  },
  {
    "pid": 15368,
    "isolate": "0x102004600",
    "time_since_init": 8049,
    "data": {
      "collector": "scavenge",
      "start_object_size": 406,
      "end_object_size": 404.3,
      "start_memory_size": 440.6,
      "end_memory_size": 452.6,
      "pause": 5.9,
      "external_time": 0,
      "step_description": "+ 6.4 ms in 42 steps since last GC",
      "gc_reason": "allocation failure"
    },
    "type": "trace"
  },
  {
    "pid": 15368,
    "isolate": "0x102004600",
    "type": "verbose",
    "data": {
      "memory_allocator": {
        "used": 463460,
        "available": 1002908
      },
      "new_space": {
        "used": 7345,
        "available": 8774,
        "committed": 32768
      },
      "old_space": {
        "used": 128565,
        "available": 9346,
        "committed": 140244
      },
      "code_space": {
        "used": 1264,
        "available": 59,
        "committed": 3072
      },
      "map_space": {
        "used": 267,
        "available": 802,
        "committed": 1104
      },
      "large_object_space": {
        "used": 276568,
        "available": 1001867,
        "committed": 283200
      },
      "all_spaces": {
        "used": 414010,
        "available": 1020849,
        "committed": 460388
      },
      "external_memory_reported": 24,
      "total_time_spent_in_gc": 128.8
    }
  },
  {
    "pid": 15368,
    "isolate": "0x102004600",
    "time_since_init": 8149,
    "data": {
      "collector": "mark_sweep",
      "start_object_size": 435.2,
      "end_object_size": 137.5,
      "start_memory_size": 480.9,
      "end_memory_size": 184.8,
      "pause": 6.4,
      "external_time": 0,
      "step_description": "+ 7.6 ms in 112 steps since start of marking, biggest step 2.0 ms",
      "gc_reason": "GC interrupt",
      "collector_reason": "GC in old space requested"
    },
    "type": "trace"
  },
  {
    "pid": 15368,
    "isolate": "0x102004600",
    "type": "verbose",
    "data": {
      "memory_allocator": {
        "used": 189252,
        "available": 1277116
      },
      "new_space": {
        "used": 3879,
        "available": 12240,
        "committed": 32768
      },
      "old_space": {
        "used": 134586,
        "available": 9839,
        "committed": 148436
      },
      "code_space": {
        "used": 1249,
        "available": 0,
        "committed": 3072
      },
      "map_space": {
        "used": 265,
        "available": 0,
        "committed": 1104
      },
      "large_object_space": {
        "used": 781,
        "available": 1276075,
        "committed": 800
      },
      "all_spaces": {
        "used": 140762,
        "available": 1298154,
        "committed": 186180
      },
      "external_memory_reported": 24,
      "total_time_spent_in_gc": 135.2
    }
  },
  {
    "pid": 15368,
    "isolate": "0x102004600",
    "time_since_init": 9874,
    "data": {
      "collector": "scavenge",
      "start_object_size": 315.6,
      "end_object_size": 314.2,
      "start_memory_size": 357.3,
      "end_memory_size": 366.3,
      "pause": 6.7,
      "external_time": 0,
      "gc_reason": "allocation failure"
    },
    "type": "trace"
  },
  {
    "pid": 15368,
    "isolate": "0x102004600",
    "type": "verbose",
    "data": {
      "memory_allocator": {
        "used": 375108,
        "available": 1091260
      },
      "new_space": {
        "used": 5428,
        "available": 10691,
        "committed": 32768
      },
      "old_space": {
        "used": 201516,
        "available": 13911,
        "committed": 219092
      },
      "code_space": {
        "used": 1287,
        "available": 974,
        "committed": 3072
      },
      "map_space": {
        "used": 267,
        "available": 802,
        "committed": 1104
      },
      "large_object_space": {
        "used": 113283,
        "available": 1090219,
        "committed": 116000
      },
      "all_spaces": {
        "used": 321784,
        "available": 1116598,
        "committed": 372036
      },
      "external_memory_reported": 16,
      "total_time_spent_in_gc": 198.4
    }
  }
]